packages/@aws-cdk/aws-events-targets/lib/sns.ts

```diff
@@ -0,0 +1,35 @@
+import events = require('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import sns = require('@aws-cdk/aws-sns');
+
+/**
+ * Use an SNS topic as a target for AWS CloudWatch event rules.
+ *
+ * @example
+ *
+ *    // publish to an SNS topic every time code is committed
+ *    // to a CodeCommit repository
+ *    repository.onCommit(new SnsTopicTarget(topic));
+ *
+ */
+export class SnsTopicTarget implements events.IEventRuleTarget {
+  constructor(public readonly topic: sns.ITopic) {
+
+  }
+
+  /**
+   * Returns a RuleTarget that can be used to trigger this SNS topic as a
+   * result from a CloudWatch event.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/resource-based-policies-cwe.html#sns-permissions
+   */
+  public asEventRuleTarget(_ruleArn: string, _ruleId: string): events.EventRuleTargetProps {
+    // deduplicated automatically
+    this.topic.grantPublish(new iam.ServicePrincipal('events.amazonaws.com'));
```

Comment:
Mmmm, lovely!

Reply:
I actually want to write a test to verify this... I am not 100% sure.